tests/unit/plugins/connection/aws_ssm/test_aws_ssm.py

```diff
@@ -307,6 +307,55 @@ def test_generate_commands(self, is_windows):
         assert isinstance(test_command_generation[0], list)
         assert isinstance(test_command_generation[0][0], dict)
 
+    @pytest.mark.parametrize(
+        "message,level,method",
+        [
+            ("test message 1", 1, "v"),
+            ("test message 2", 2, "vv"),
+            ("test message 3", 3, "vvv"),
+            ("test message 4", 4, "vvvv"),
+        ],
+    )
+    def test_verbosity_diplay(self, message, level, method):
+        """Testing verbosity levels"""
+        play_context = MagicMock()
+        play_context.shell = "sh"
+        conn = Connection(play_context)
+        conn.host = "test-host"  # Test with host set
+
+        with patch("ansible_collections.community.aws.plugins.connection.aws_ssm.display") as mock_display:
+            conn.verbosity_display(level, message)
+            # Verify the correct display method was called with expected args
+            mock_method = getattr(mock_display, method)
+            mock_method.assert_called_once_with(message, host="test-host")
+
+            # Test without host set
+            conn.host = None
+            conn.verbosity_display(1, "no host message")
+            mock_display.v.assert_called_with("no host message")
+
+    def test_poll_verbosity(self):
+        """Test poll method verbosity display"""
+        pc = PlayContext()
+        new_stdin = StringIO()
+        conn = connection_loader.get("community.aws.aws_ssm", pc, new_stdin)
+
+        conn._session = MagicMock()
+        conn._session.poll.return_value = None
+        conn.get_option = MagicMock(return_value=10)  # ssm_timeout
+        conn.poll_stdout = MagicMock()
+        conn.instance_id = "i-1234567890"
+        conn.host = conn.instance_id
+
+        with patch("time.time", return_value=100), patch.object(conn, "verbosity_display") as mock_display:
+            poll_gen = conn.poll("TEST", "test command")
+            # Advance generator twice to trigger the verbosity message
+            next(poll_gen)
+            next(poll_gen)
+
+            # Verify verbosity message contains remaining time
+            mock_display.assert_called_with(4, "TEST remaining: 10 second(s)")
```

Comment:
I figured it might be a good way to test `verbosity_display()` by utilizing a method that calls `verbosity_display()`, but maybe this is gonna cause more trouble than it's worth since the `poll()` method might move in the future.